### PR TITLE
[Feature]update memory size code for docker container

### LINKED ISSUE #9729: [Enhancement] Docker instance gets the wrong memory limit
## Enhancement

When deploying StarRocks with Docker, the total memory obtained is the memory of the host machine, which should be the memory limit of the Docker instance itself.

### PR DESCRIPTION
**Pull request by dufeng1010**
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

fix #9729 

## Problem Summary(Required) ：
when starrocks run in docker, it get host's memory, not docker memory limit.
starrocks run in docker, it need memory limit via /sys/fs/cgroup/memory/memory.limit_in_bytes file.
## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

File: be/src/util/mem_info.cpp [hunk 1]
```diff
@@ -35,36 +35,58 @@ bool MemInfo::_s_initialized = false;
 int64_t MemInfo::_s_physical_mem = -1;
 
 void MemInfo::init() {
-    // Read from /proc/meminfo
-    std::ifstream meminfo("/proc/meminfo", std::ios::in);
-    std::string line;
+    // check if application is in docker container or not via /.dockerenv file
+    std::ifstream dockerEnv("/.dockerenv");
```
Comment (imay) [inline be/src/util/mem_info.cpp:39]: you can use `path_exist` to check if this path exists, which locates `be/src/fs/fs_util.h`

Comment (dufeng1010) [reply be/src/util/mem_info.cpp:39]: @imay ok, i will use path_exist function in be/src/fs/fs_util.h file.

File: be/src/util/mem_info.cpp [hunk 2]
```diff
@@ -35,36 +35,58 @@ bool MemInfo::_s_initialized = false;
...
 
-    while (meminfo.good() && !meminfo.eof()) {
-        getline(meminfo, line);
-        std::vector<std::string> fields = strings::Split(line, " ", strings::SkipWhitespace());
+    if (dockerEnv.goods() && !dockerEnv.eof()) {
```
Comment (trueeyu) [inline be/src/util/mem_info.cpp:41]: goods -> good

**Review by trueeyu — DISMISSED** [review]

**Review by trueeyu — DISMISSED** [review]

**Review by trueeyu — APPROVED** [review]

**Review by imay — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by dengliu** [comment]
This cause start_be.sh failure
```
# ./start_be.sh --daemon
cat: /sys/fs/cgroup/memory/memory.limit_in_bytes: No such file or directory
can't get mem info from /sys/fs/cgroup/memory/memory.limit_in_bytes
```

There is not `/sys/fs/cgroup/memory/memory.limit_in_bytes` file for a centos-7 base docker container.
E.g, running this command in a centos-7 based docker container got this error.

```
# ls /sys/fs/cgroup/memory
ls: cannot access /sys/fs/cgroup/memory: No such file or directory

```

**Pull request merged**